packages/iov-bcp-types/src/transactions.ts

```diff
@@ -21,8 +21,26 @@ export type SwapIdString = string & As<"swap-id">;
 export type RecipientId = Address;
 
 export interface Amount {
-  readonly whole: number;
-  readonly fractional: number;
+  /**
+   * The quantity expressed as atimic units.
+   *
+   * Convert to whole and fractional part using
+   *   const whole = amount.quantity.slice(0, -amount.fractionalDigits);
+   *   const fractional = amount.quantity.slice(-amount.fractionalDigits);
+   * or to a floating point approximation (not safe!)
```

Comment:
Because you need to draw the line somewhere. When integer string is the library interface, it should be the thing the library exposes. There are at least 3 different ways in here to parse the result and all require some kind of big int library. Also I expect users to have their own ways. Especially because every node10+ caller can use native BigInteger, which we can't supporting node 8.

The comment aims to explain how to understand the result.

Reply:
If I was a caller I wanted to have a BigDecimal class that does not only parse and store but also do math